src/AWS.Deploy.Orchestration/ElasticBeanstalkHandler.cs

```diff
@@ -0,0 +1,201 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Amazon.ElasticBeanstalk;
+using Amazon.ElasticBeanstalk.Model;
+using AWS.Deploy.Common;
+using AWS.Deploy.Common.IO;
+
+namespace AWS.Deploy.Orchestration
+{
+    public interface IElasticBeanstalkHandler
+    {
+        Task<S3Location> CreateApplicationStorageLocationAsync(string applicationName, string versionLabel, string deploymentPackage);
+        Task<CreateApplicationVersionResponse> CreateApplicationVersionAsync(string applicationName, string versionLabel, S3Location sourceBundle);
+        Task<bool> UpdateEnvironmentAsync(string applicationName, string environmentName, string versionLabel, List<ConfigurationOptionSetting> optionSettings);
+        List<ConfigurationOptionSetting> GetAdditionalSettings(Recommendation recommendation);
```

Comment:
What makes these settings "Additional"? Either come up with a better name or add docs to the method.